Initial support for AWS Batch Job Queues

jonmer85 opened the pull request:
SUMMARY

This adds automation support for the following AWS Batch Job Queues
ISSUE TYPE

Feature Pull Request
COMPONENT NAME

amazon module

ANSIBLE VERSION

ansible 2.2.2.0
config file =
configured module search path = Default w/o overrides

jonmeransonos committed:
Initial support for AWS Batch Job Queues

ryansb commented on lib/ansible/modules/cloud/amazon/batch_job_queue.py:
```diff
@@ -0,0 +1,384 @@
+
+DOCUMENTATION = '''
+---
+module: batch_job_queue
```
Same - `aws_batch_job_queue`

supreethSonos replied:
fixed

supreethSonos committed:
Changes include:
1. Removed redundant default values in argument_spec
2. Enclosed non standard python libraries in Try Except ImportError block
3. Added region as a mandatory element in argument_spec
4. Updated the typo in api_params while accessing job queue. Changed it from job_queue to jobQueue
5. Updated the set_api_params() method to use snake_dict_to_camel_dict()
6. Updated the ANSIBLE_METADATA version from 1.0 to 1.1
7. Updated the version_added in Documentation from 2.4 to 2.5
8. Initialized response value to None
9. Updated the RETUN documentation
10. Updated the return from manage_state() method to not embed batch_job_queue_action in ansible_facts
11. Updated the module.exit_json()  in main() to convert any keys back to snake_dict format using camel_dict_to_snake_dict to maintain consistency throughout the module
12. Changed the filename from batch_job_queue.py to aws_batch_job_queue.py

supreethSonos committed:
Fixed Botocore error

s-hertel committed:
Update copyright

Remove exception handling around boto3_conn as it is handled in it

Add ec2 documentation fragment

Document supports_check_mode

Remove HAS_BOTOCORE since HAS_BOTO3 handles it

Use new style class

Region is not required because it may be provided as an environment variable

Add aliases for region

ryansb commented on lib/ansible/modules/cloud/amazon/aws_batch_job_queue.py:
```diff
@@ -0,0 +1,384 @@
+#!/usr/bin/python
+# Copyright (c) 2017 Jon Meran <jonathan.meran@sonos.com>
```
Needs to follow copyright guidelines http://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#copyright

s-hertel committed:
Import shared code
